appcues/src/main/java/com/appcues/data/remote/NetworkRequest.kt

```diff
@@ -0,0 +1,27 @@
+package com.appcues.data.remote
+
+import com.appcues.data.MoshiConfiguration
+import com.appcues.data.remote.appcues.response.ErrorResponse
+import com.appcues.util.ResultOf
+import com.squareup.moshi.JsonDataException
+import retrofit2.HttpException
+
+object NetworkRequest {
+    internal suspend fun <Success> execute(apiCall: suspend () -> Success): ResultOf<Success, RemoteError> =
+        try {
+            ResultOf.Success(apiCall.invoke())
+        } catch (exception: HttpException) {
+            ResultOf.Failure(RemoteError.HttpError(exception.code(), convertErrorBody(exception)))
+        } catch (exception: Exception) {
+            ResultOf.Failure(RemoteError.NetworkError(exception))
+        }
```

Comment:
this helper for error handling was in the `RetrofitAppcuesRemoteSource` previously - moved out to this shared spot so that it could be used by all the remote sources making network calls now.